datawig/utils.py

```diff
@@ -302,4 +318,53 @@ def generate_df_numeric(num_samples: int = 100,
     return pd.DataFrame({
         data_column_name: numeric_data,
         label_column_name: numeric_data ** 2 + np.random.normal(0, .01, (num_samples,)),
-    })
\ No newline at end of file
+    })
+
+
+def random_cartesian_product(sets, num=10):
+    """
+    Return num random samples from the cartesian product of all iterators in sets.
+    Returns at most as many results as unique products exist.
+    """
```

Comment:
Please make this a standard docstring, with `:param ...`. An example input/output of this function would help understand it better.